build/configs/imxrt1020-evk/imxrt1020-evk_download.sh

```diff
@@ -18,34 +18,167 @@
 ###########################################################################
 # imxrt1020-evk_download.sh
 
+WARNING="\n########################################\n
+	WARNING:Make sure NO other application like \n
+	putty etc, is accessing ttyACM0. If download \n
+	fails, reset the board before next try!!\n
+	#################################################\n"
 
-THIS_PATH=`test -d ${0%/*} && cd ${0%/*}; pwd`
+echo -e $WARNING
+
+
+CURDIR=$(readlink -f "$0")
+CURDIR_PATH=$(dirname "$CURDIR")
 
 # When location of this script is changed, only TOP_PATH should be changed together!!!
-TOP_PATH=${THIS_PATH}/../../..
+TOP_PATH=${CURDIR_PATH}/../../..
 
 OS_PATH=${TOP_PATH}/os
 OUTBIN_PATH=${TOP_PATH}/build/output/bin
+TTYDEV="/dev/ttyACM0"
 
 CONFIG=${OS_PATH}/.config
-if [ ! -f ${CONFIG} ]; then
+if [ ! -f ${CONFIG} ]
+then
         echo "No .config file"
         exit 1
 fi
 
 source ${CONFIG}
-source ${THIS_PATH}/imxrt1020-evk_get_filename.sh
 
-IMXRT1020_USB_STRING=`mount | grep RT1020-EVK`
-FIND_WORD="RT1020-EVK"
+if [[ "${CONFIG_ARCH_BOARD_IMXRT1020_EVK}" != "y" ]]
+then
+	echo "Target is NOT IMXRT1020_EVK"
+	exit 1
+fi
+
+if [ ! -f ${TINYARA_BIN} ]; then
```

Comment:
Where is `TINYARA_BIN` definition?

Reply:
Updated with definition. Thanks for the catch.

Reply:
I think you have to get the path of tinyara bin.
```
if [ ! -f ${OUTBIN_PATH}/${TINYARA_BIN} ]; then

or

TINYARA_BIN="${OUTBIN_PATH}/tinyara.bin"
```

Reply:
Oh yes, have to be fixed and tested more.

Reply:
With latest fix:
########################################
 WARNING:Make sure NO other application like 
 putty etc, is accessing ttyACM0. If download 
 fails, reset the board before next try!!
 #################################################

missing file /home/bsvt18/TizenRT/PublicGit/June6/build/configs/imxrt1020-evk/../../../build/output/bin/tinyara.bin
Makefile.unix:533: recipe for target 'download' failed
make: *** [download] Error 1